frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -154,20 +164,34 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    this.setState({ breadcrumbsList: breadcrumbsList })
+    return breadcrumbsList
   }
 
   loadFolderDetail = async (workspaceId, folderId) => {
     const { state, props } = this
 
-    const fetchContentDetail = await handleFetchResult(
+    let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        this.setState({ fileName: fetchContentDetail.body.filename })
-        break
+        const fileName = fetchContentDetail.body.filename
+        let folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+
+        while (fetchContentDetail.body.parent_id !== null) {
+          fetchContentDetail = await handleFetchResult(
+            await getFolderDetail(state.config.apiUrl, workspaceId, fetchContentDetail.body.parent_id)
+          )
+          if (fetchContentDetail.apiResponse.status === 200) {
+            folderParentsId.push(fetchContentDetail.body.parent_id)
+          } else {
+            this.sendGlobalFlashMessage(props.t('Error while loading folder detail'))
```

Comment:
this error case can spam.
Your while variable can end up in an infinite loop.
You need to handle your iteration variable manually depending on `fetchContentDetail`. Don't depend on the result of another function. Even more so that the other function directly depends on what the api does

Reply:
@metzgegu I think you did not fix this. If the api doesn't returns http status 200, your iteration variable wont be updated and might spam api calls

Reply:
@metzgegu still not fixed. Your function can still return undefined in the `default` case